File: getData.py
from web3 import Web3
from web3.middleware import geth_poa_middleware
import json
import time
import datetime

my_provider = Web3.HTTPProvider('http://137.116.200.211:22000')
w3 = Web3(my_provider)
w3.middleware_stack.inject(geth_poa_middleware, layer=0)
count = w3.eth.blockNumber

for i in range(count): 
    block = w3.eth.getBlock(i)
    timeStamp = block.timestamp

    value = time.strftime('%m/%d/%Y %H:%M:%S',  time.gmtime(timeStamp/1000000.))
    # value = datetime.datetime.fromtimestamp(timeStamp)
    # exct_time = value.strftime('%d %B %Y %H:%M:%S')

    # dateTime = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(timeStamp / 1000000))
    txnCount = w3.eth.getBlockTransactionCount(i)
    for j in range(txnCount): 
        if ( j==1 ): 
            print(timeStamp) 
            print(value)
        

File: getData.js
var Web3 = require('web3');
var json2xls = require('json2xls');
var fs = require('fs');

var provider = "http://localhost:22000";
var web3;

var txnData = [];
var count;


if (typeof web3 !== 'undefined') {
    web3 = new Web3(web3.currentProvider);
} else {
    // set the provider you want from Web3.providers
    web3 = new Web3(new Web3.providers.HttpProvider(provider));
}


(async function getData() {
  count =  web3.eth.blockNumber;
  try {
    //Get Transaction count
    await getTxn();
 
    console.log("Completed Successfully!!!");

  } catch (err) {
    console.log(err);
  }

})()

async function getTxn() {
  var start = 0;
  for (var i = start; i <= count; i++) {
    if(i%500 == 0 || i == start){
      console.log("Writing to file......");
    }
    
    var error, result = await web3.eth.getBlockTransactionCount(i);
    if(!error){
        var err, blkData = await web3.eth.getBlock(i);
        if(!err){

          var unix_timestamp = blkData.timestamp / 1000000;
          var date = new Date(unix_timestamp);
         
          var convertedTimeStamp = ("00" + date.getDate()).slice(-2) + "/" + 
                                   ("00" + (date.getMonth() + 1)).slice(-2) + "/" + 
                                   date.getFullYear() + " " + 
                                   ("00" + date.getHours()).slice(-2) + ":" + 
                                   ("00" + date.getMinutes()).slice(-2) + ":" + 
                                   ("00" + date.getSeconds()).slice(-2);             

          var res = {
            "block": i,
            "transactionCount": result,
            "time": convertedTimeStamp,
        }

         txnData.push(res);
        }
    }
    
    if((i%500 == 0 && i != 0) || i == count){

      var xls = json2xls(txnData);
      fs.writeFileSync('data' + i + '.xlsx', xls, 'binary');
      console.log("written upto " + i);
      txnData = [];

    }

  }
}